app/models/user_link.rb

```diff
@@ -1,3 +1,14 @@
 class UserLink < ActiveRecord::Base
+  before_save :fix_link
   belongs_to :user
+  scope :with_link, ->{ where('link IS NOT NULL') }
+
+  def link_without_protocol
+    self.link.sub(/^https?\:\/\//, '').sub(/^www./,'')
+  end
+
+  def fix_link
```

Comment:
Let's rename this to something more explanatory, like prepend_protocol, or infer_protocol.